project/crud.go

```diff
@@ -23,15 +25,39 @@ func Clone(args []string) error {
 }
 
 func Delete(args []string) error {
+	// only works with company management key, ignore errors other than project not found
+	projects, err := shared.Descope.Management.Project().ListProjects(context.Background())
+	if err, ok := err.(*descope.Error); ok && err.Code == errProjectNotFound {
+		return fmt.Errorf("cannot find project with id %s", args[0])
+	}
+
+	alias := args[0]
+	name := ""
+	for _, p := range projects {
+		if p.ID == args[0] {
```

Comment:
In case we don't find the project in the list, shouldn't we raise the error early instead of doing the call?

Reply:
The check above should catch that. This loop needs to handle the case where the `ListProjects` call cannot be made because the management key is not a company-wide one.